src/awscdk/codecatalyst.ts

```diff
@@ -0,0 +1,342 @@
+import { Environment } from '@amazon-codecatalyst/blueprint-component.environments';
+import { convertToWorkflowEnvironment, WorkflowBuilder } from '@amazon-codecatalyst/blueprint-component.workflows';
+import { YamlFile, awscdk } from 'projen';
+import { CDKPipeline, CDKPipelineOptions, DeploymentStage } from './base';
+
+import { Blueprint } from './codecatalyst/blueprint';
+import { PipelineEngine } from '../engine';
+import { PipelineStep, SimpleCommandStep, UploadArtifactStep } from '../steps';
+
+/*
+Needs to create:
+- build.yml (creates and mutates pipeline by executing projen build) - comes from projen
+- deploy.yaml (build + deploy to dev)
+- pull-request-lint.yml (executes amannn/action-semantic-pull-request@v5.0.2) - comes from projen
+- release-prod.yaml (deploy to prod - not required, move over to "manual approvals" in deplo)
+- upgrade.yaml (upgrade dependencies)  - comes from projen
+
+* synth -> create artifacts
+* upload cdk assets ->  save assets in s3 (lambda), build container images (push to ECR) -- everything in AWS
+* deploy for each stage that is non-production
+* deploy to prod (manual approval)
+
+TODO:
+- account target -- NOT POSSIBLE as we cannot create environments/accounts/targets
+- manual approval for stages -- DONE
+- IAM role per stage, synth, asset - NOT POSSIBLE as we cannot create environments
+- independend stages (all parallel to each other) after synth&assets -- DONE
+- environments support - DONE
+- steps per stage - preInstall, preSynth, ... - DONE
+
+example: https://github.com/aws-community-dach/event-system-backend
+
+test docgen: https://github.com/open-constructs/aws-cdk-library
+
+
+*/
+
+export interface CodeCatalystIamRoleConfig {
```

Comment:
Cool, given that we need it for all engines and it behaves the same, I will refactor this once merged to the base class